src/Modules/Visualization/ShowField.cc

```diff
@@ -430,6 +430,96 @@ void GeometryBuilder::renderFaces(
 
 
 
+inline static void writeAtributeToVBO(const Point& point, spire::VarBuffer* vboBuffer)
+{
+  vboBuffer->write(static_cast<float>(point.x()));
+  vboBuffer->write(static_cast<float>(point.y()));
+  vboBuffer->write(static_cast<float>(point.z()));
+}
+
+inline static void writeAtributeToVBO(const Vector& vector, spire::VarBuffer* vboBuffer)
+{
+  vboBuffer->write(static_cast<float>(vector.x()));
+  vboBuffer->write(static_cast<float>(vector.y()));
+  vboBuffer->write(static_cast<float>(vector.z()));
+}
+
+inline static void writeAtributeToVBO(const ColorRGB& color, spire::VarBuffer* vboBuffer)
+{
+  vboBuffer->write(static_cast<float>(color.r()));
+  vboBuffer->write(static_cast<float>(color.g()));
+  vboBuffer->write(static_cast<float>(color.b()));
+  vboBuffer->write(static_cast<float>(1.0f));
+}
+
+inline static void writeAtributeToVBO(const glm::vec2& coords, spire::VarBuffer* vboBuffer)
+{
+  vboBuffer->write(static_cast<float>(coords.x));
+  vboBuffer->write(static_cast<float>(coords.y));
+}
+
+inline static void writeIndexToIBO(const uint32_t index, spire::VarBuffer* iboBuffer)
+{
+  iboBuffer->write(index);
+}
+
+template<typename ... Params>
+static void writeTri(
+  spire::VarBuffer* vboBuffer,
+  spire::VarBuffer* iboBuffer,
+  uint32_t& iboIndex,
+  const Params& ... params)
+{
+  for(int i = 0; i < 3; ++i)
+    (void)std::initializer_list<int>{(writeAtributeToVBO(params[i], vboBuffer), 0)...};
+
+  writeIndexToIBO(iboIndex + 0, iboBuffer);
+  writeIndexToIBO(iboIndex + 1, iboBuffer);
+  writeIndexToIBO(iboIndex + 2, iboBuffer);
+  iboIndex += 3;
+}
+
+template<typename ...Params>
+static void writeQuad(
+  spire::VarBuffer* vboBuffer,
+  spire::VarBuffer* iboBuffer,
+  uint32_t& iboIndex,
+  const Params& ... params)
+{
+  for(int i = 0; i < 4; ++i)
+    (void)std::initializer_list<int>{(writeAtributeToVBO(params[i], vboBuffer), 0)...};
+
+  writeIndexToIBO(iboIndex + 0, iboBuffer);
+  writeIndexToIBO(iboIndex + 1, iboBuffer);
+  writeIndexToIBO(iboIndex + 2, iboBuffer);
+  writeIndexToIBO(iboIndex + 2, iboBuffer);
+  writeIndexToIBO(iboIndex + 3, iboBuffer);
+  writeIndexToIBO(iboIndex + 0, iboBuffer);
+  iboIndex += 4;
+}
+
+
+
+static float valueToFloat(double scalar)
```

Comment:
These color conversions need to be exposed and better documented. Could you extract a class, say, `ShowFieldDataColorConverter` which contains these 3 functions?

Reply:
I actually found a better way to do it so those are unused now. I will remove them